### PR TITLE
docs(x/ecocredit): update client docs

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Ref: #1456 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by ryanchristo — APPROVED** [review]
LGTM

File: x/ecocredit/spec/07_client.md [hunk 1]
```diff
@@ -1735,6 +1933,177 @@ Example Output:
...
+
+```bash
+curl localhost:1317/regen/ecocredit/v1/allowed-class-creators
+```
+
+Example Output:
+
+```bash
+{
+  "class_creators":[
```
Comment (technicallyty) [inline x/ecocredit/spec/07_client.md:2047]: noticing that some of these are snake_case and others are camelCase - are some of these outdated?

Comment (ryanchristo) [reply x/ecocredit/spec/07_client.md:2047]: rpc returns camel case and rest returns snake case

Comment (ryanchristo) [reply x/ecocredit/spec/07_client.md:2047]:
```
grpcurl -plaintext -d '{"batch_denom": "C01-001-20200101-20210101-001"}' localhost:9090 regen.ecocredit.v1.Query/BalancesByBatch

...

{
  "balances": [
    {
      "address": "regen1xm4ty25m95gvv6h7exr65luh6uv3fqyypnmgmw",
      "batchDenom": "C01-001-20200101-20210101-001",
      "tradableAmount": "1000",
      "retiredAmount": "500",
      "escrowedAmount": "0"
    }
  ],
  "pagination": {
    "total": "1"
  }
}
```

```
curl localhost:1317/regen/ecocredit/v1/batches/C01-001-20200101-20210101-001/balances

...

{
  "balances": [
    {
      "address": "regen1xm4ty25m95gvv6h7exr65luh6uv3fqyypnmgmw",
      "batch_denom": "C01-001-20200101-20210101-001",
      "tradable_amount": "1000",
      "retired_amount": "500",
      "escrowed_amount": "0"
    }
  ],
  "pagination": {
    "next_key": null,
    "total": "1"
  }
}
```

**Pull request merged**